packages/flutter/test/material/expansion_tile_test.dart

```diff
@@ -229,4 +229,93 @@ void main() {
   });
+
+  testWidgets('ExpansionTile custom tween colors', (WidgetTester tester) async {
+    final Key iconKey = UniqueKey();
+    final Key titleKey = UniqueKey();
+    final ColorTween headerColorTween = ColorTween(
+      begin: Colors.red,
+      end: Colors.blue,
+    );
+    final ColorTween iconColorTween = ColorTween(
+      begin: Colors.orange,
+      end: Colors.green,
+    );
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Material(
+          child: ExpansionTile(
+            initiallyExpanded: true,
+            headerColorTween: headerColorTween,
+            iconColorTween: iconColorTween,
+            title: TestText('Expanded', key: titleKey),
+            trailing: TestIcon(key: iconKey),
+          ),
+        ),
+      ),
+    );
+
+    Color iconColor(Key key) => tester.state<TestIconState>(find.byKey(key)).iconTheme.color;
+    Color textColor(Key key) => tester.state<TestTextState>(find.byKey(key)).textStyle.color;
+
+    expect(iconColor(iconKey), Colors.green);
+    expect(textColor(titleKey), Colors.blue);
+
+    // Tap the tile to change its state to collapsed
+    await tester.tap(find.text('Expanded'));
+    await tester.pump();
+    await tester.pump(const Duration(seconds: 1));
+    await tester.pump(const Duration(seconds: 1));
+
+    expect(iconColor(iconKey), Colors.orange);
+    expect(textColor(titleKey), Colors.red);
+  });
+
+  testWidgets('ExpansionTile mixed custom and default tween colors', (WidgetTester tester) async {
+    final Key iconKey = UniqueKey();
+    final Key titleKey = UniqueKey();
+    final ColorTween headerColorTween = ColorTween(
+      begin: Colors.red,
+    );
+    final ColorTween iconColorTween = ColorTween(
+      end: Colors.green,
+    );
+    /// ThemeData.accentColor is used as the default value for headerColorTween.end
+    /// ThemeData.unselectedWidgetColor is used as the default value for iconColorTween.begin
+    final ThemeData theme = ThemeData(
+      accentColor: Colors.blue,
+      unselectedWidgetColor: Colors.orange,
+    );
+
+    await tester.pumpWidget(
+      MaterialApp(
+        theme: theme,
+        home: Material(
+          child: ExpansionTile(
+            initiallyExpanded: true,
+            headerColorTween: headerColorTween,
+            iconColorTween: iconColorTween,
+            title: TestText('Expanded', key: titleKey),
+            trailing: TestIcon(key: iconKey),
+          ),
+        ),
+      ),
+    );
+
+    Color iconColor(Key key) => tester.state<TestIconState>(find.byKey(key)).iconTheme.color;
+    Color textColor(Key key) => tester.state<TestTextState>(find.byKey(key)).textStyle.color;
+
+    expect(iconColor(iconKey), Colors.green);
+    expect(textColor(titleKey), Colors.blue);
+
+    // Tap the tile to change its state to collapsed
+    await tester.tap(find.text('Expanded'));
+    await tester.pump();
+    await tester.pump(const Duration(seconds: 1));
+    await tester.pump(const Duration(seconds: 1));
```

Comment:
Why are there two pump calls with a duration of 1 second for each here?

Reply:
Honestly, I have no idea :)

This test case is based on the `ListTileTheme` from the same file. I tried to keep the same structure. I always try to do so and change if requested 😅

I tried to remove one of these during the last refactor but wasn't able to do so, tests were failing. But thanks for bringing it up here, I checked other tests and replaced these `.pump` calls with `.pumpAndSettle`. Hope this is a good practice - I'm still quite new to testing in Flutter/Dart, have just started with unit tests.

Reply:
I did some additional digging, and it seems like there is an issue with trying to update the header color. This works:
```dart
    final Key iconKey = UniqueKey();
    final Key titleKey = UniqueKey();
    final ColorTween headerColorTween = ColorTween(
      begin: Colors.red,
    );
    final ColorTween iconColorTween = ColorTween(
      end: Colors.green,
    );
    /// ThemeData.accentColor is used as the default value for headerColorTween.end
    /// ThemeData.unselectedWidgetColor is used as the default value for iconColorTween.begin
    final ThemeData theme = ThemeData(
      accentColor: Colors.blue,
      unselectedWidgetColor: Colors.orange,
    );

    await tester.pumpWidget(
      MaterialApp(
        theme: theme,
        home: Material(
          child: ExpansionTile(
            initiallyExpanded: true,
            headerColorTween: headerColorTween,
            iconColorTween: iconColorTween,
            title: TestText('Expanded', key: titleKey),
            trailing: TestIcon(key: iconKey),
          ),
        ),
      ),
    );

    Color iconColor(Key key) => tester.state<TestIconState>(find.byKey(key)).iconTheme.color;
    expect(iconColor(iconKey), Colors.green);

    // Tap the tile to change its state to collapsed
    await tester.tap(find.text('Expanded'));
    await tester.pump();
    await tester.pump(const Duration(milliseconds: 100));

    // The icon's color transitions with an ease in curve.
    expect(iconColor(iconKey), const Color(0xffc69f19));
    await tester.pump(const Duration(milliseconds: 100));

    // Fully transitions to end color.
    expect(iconColor(iconKey), Colors.orange);
```

However, tweaking the numbers on `textColor`, it seems that it takes 400ms or longer for the color changes to reach the endpoint and it has intermediary colors that didn't make sense to me. My hunch is that there is some strange interaction between this tween and some underlying animation mechanism.

I think we should reconsider this change until we understand why this is happening. Does that sound okay with you?

Reply:
I thought about adding just the icon tween, but am a little concerned about that idea since we don't understand why that one works but the header tween does not.

Reply:
I'd like to have the icon as customizable but I completely understand your points and I have a workaround in place. Feel free to close the PR from my side. 👍